Gemfile.lock

```diff
@@ -354,17 +354,9 @@ GEM
       connection_pool (~> 2.2)
     nio4r (2.5.4)
     nio4r (2.5.4-java)
-    nokogiri (1.11.1)
+    nokogiri (1.11.3)
       mini_portile2 (~> 2.5.0)
       racc (~> 1.4)
-    nokogiri (1.11.1-java)
-      racc (~> 1.4)
-    nokogiri (1.11.1-x64-mingw32)
-      racc (~> 1.4)
-    nokogiri (1.11.1-x86-mingw32)
-      racc (~> 1.4)
-    nokogiri (1.11.1-x86_64-darwin)
-      racc (~> 1.4)
```

Comment:
I'm not sure why these other architectures were removed 🤔

Reply:
I think this is expected behavior since nokogiri gem is now precompiled native gem since 1.11.0. Refer https://github.com/sparklemotion/nokogiri/issues/2075 and https://github.com/sparklemotion/nokogiri/issues/2185 From Rails side, what we can do now is waiting for a new version of nokogiri gem after Ruby 3.1.0 is released.

Also, Rails CI actually removes `Gemfile.lock` then `bundle install` .

https://github.com/rails/buildkite-config/blob/8b7d116453fa6fcf3075e8a685686ac5b69e2fcb/Dockerfile#L133
```
    && rm Gemfile.lock && bundle install -j 8 && cp Gemfile.lock tmp/Gemfile.lock.updated \
```
So, unless we need to bump gem version at `Gemfile` itself, we do not need to update `Gemfile.lock` to support Ruby 3.1.0-dev.

Reply:
@yahonda Hello!

> this is expected behavior since nokogiri gem is now precompiled native gem since 1.11.0. Refer sparklemotion/nokogiri#2075 and sparklemotion/nokogiri#2185

I understand that the behavior of nokogiri is to install the precompiled gem for my architecture, but I'm wondering why just bumping from `1.11.1` to `1.11.3` would remove the other architectures from the lock file. Maybe its my version of bundler from ruby HEAD?

It also seems that only [this version on `x86_64-darwin`](https://rubygems.org/gems/nokogiri/versions/1.11.2-x86_64-darwin) is affected.

Would it be a big deal to just merge these versions which are working locally? I assume they don't break for existing Ruby versions, but the alternative is having to maintain a dirty `Gemfile.lock` in my local checkout in order to hack on Rails 🤔

Reply:
I think this is a Bundler issue: https://github.com/rubygems/rubygems/issues/4516